### ISSUE TITLE
8.06.14: build fails

### ISSUE THREAD

**Issue by kloczek** [garrigue/labltk#24]
With latest glibc when in CFLAGS are used `-Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3` some warnings are turned into errors
```console
/usr/bin/gcc  -O2 -fno-strict-aliasing -fwrapv -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -O2 -g -grecord-gcc-switches -pipe -mtls-dialect=gnu2 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fdata-sections -ffunction-sections -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -flto=auto -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -Wall -Werror=format-security -pthread -fPIC -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -O2 -g -grecord-gcc-switches -pipe -mtls-dialect=gnu2 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fdata-sections -ffunction-sections -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -flto=auto -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -Wall -Werror=format-security -pthread -I/usr/lib64/ocaml/caml  -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fPIC -g -c cltkMain.c
cltkMain.c: In function ‘camltk_opentk’:
cltkMain.c:92:23: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   92 |       value *interp = caml_named_value("cltclinterp");
      |                       ^~~~~~~~~~~~~~~~
cltkMain.c:116:16: error: assignment to ‘const char **’ from incompatible pointer type ‘char **’ [-Wincompatible-pointer-types]
  116 |         tkargv = (char**)caml_stat_alloc(sizeof( char* ) * argc );
      |                ^
make[1]: *** [Makefile:89: cltkMain.o] Error 1
```

**Comment by garrigue** [comment]
Can you clarify, who is setting `-U_FORTIFY_SOURCE` ?
Does it mean that you cannot install through opam, requiring an immediate fix, or just that these should be fixed for the future ?

**Comment by kloczek** [comment]
https://www.redhat.com/en/blog/enhance-application-security-fortifysource

**Comment by garrigue** [comment]
OK, this is just a desirable improvement. I have committed a fix (f335ec6059), but it will only be available in the next release.

**Comment by garrigue** [comment]
Actually, I updated the release, since some distributions seem to be extra strict on this.
